dask/bytes/tests/test_s3.py

```diff
@@ -480,6 +480,60 @@ def test_parquet(s3, engine, s3so, metadata_file):
     tm.assert_frame_equal(data, df2.compute())
 
 
+@pytest.mark.parametrize("engine", ["pyarrow", "fastparquet"])
+def test_parquet_append(s3, engine, s3so):
+    pytest.importorskip(engine)
+    dd = pytest.importorskip("dask.dataframe")
+    pd = pytest.importorskip("pandas")
+    np = pytest.importorskip("numpy")
+    from dask.dataframe._compat import tm
+
+    url = "s3://%s/test.parquet.append" % test_bucket_name
+
+    data = pd.DataFrame(
+        {
+            "i32": np.arange(1000, dtype=np.int32),
+            "i64": np.arange(1000, dtype=np.int64),
+            "f": np.arange(1000, dtype=np.float64),
+            "bhello": np.random.choice(["hello", "you", "people"], size=1000).astype(
+                "O"
+            ),
+        },
+    )
+    df = dd.from_pandas(data, chunksize=500)
+    df.to_parquet(
+        url,
+        engine=engine,
+        storage_options=s3so,
+        write_index=False,
+    )
+    df.to_parquet(
+        url,
+        engine=engine,
+        storage_options=s3so,
+        write_index=False,
+        append=True,
+        ignore_divisions=True,
+    )
+
+    files = [f.split("/")[-1] for f in s3.ls(url)]
+    assert "_common_metadata" in files
+    assert "_metadata" in files
+    assert "part.0.parquet" in files
+
+    df2 = dd.read_parquet(
+        url,
+        index=False,
+        engine=engine,
+        storage_options=s3so,
+    )
+
+    tm.assert_frame_equal(
+        pd.concat([data, data], ignore_index=True),
+        df2.compute().reset_index(drop=True),
+    )
```

Comment:
I'm curious why we're using `tm.assert_frame_equal` here over the usual `assert_eq`?

Reply:
Yeah, fair question.  I was just copying the existing test_parquet test. I did want to update the tests a bit, but I am having trouble getting s3/moto to work in my local environment (so I am currently relying on CI more than I'd like to).

Reply:
Hmm interesting, do you have a traceback or something you could share?

Since we're mocking with `moto` I don't _think_ any additional setup is needed beyond having `moto`, `s3fs`, etc. installed. How are you setting up your local dev environment? For example, using https://github.com/dask/dask/blob/9f5875070d4986b4f6eac6f0cf5b05f8b28e17e9/continuous_integration/environment-3.9.yaml should install all the packages you need for testing

Reply:
Check for a moto process that was not cleared up from a previous test. This happens to me very occasionally.

Reply:
I crated my environment from `environment-3.8.yaml`, and i get the folowing failure for most tests:

```
                timeout -= 0.1
                time.sleep(0.1)
>               assert timeout > 0, "Timed out waiting for moto server"
E               AssertionError: Timed out waiting for moto server
E               assert -0.09999999999998832 > 0

dask/bytes/tests/test_s3.py:93: AssertionError
```

Reply:
Suggests you might have something using the port moto is trying on, 5555 (either an old moto or something else). The wait loop should probably check the status of the background process, and report any error it has to say (which might appear on your console if you sun pytest with `-s`).

Reply:
Ah - Yup:

```
dask/bytes/tests/test_s3.py::test_parquet[True-pyarrow] Traceback (most recent call last):
  File "/datasets/rzamora/miniconda3/envs/debug_8072/bin/moto_server", line 10, in <module>
    sys.exit(main())
  File "/datasets/rzamora/miniconda3/envs/debug_8072/lib/python3.8/site-packages/moto/server.py", line 316, in main
    run_simple(
  File "/datasets/rzamora/miniconda3/envs/debug_8072/lib/python3.8/site-packages/werkzeug/serving.py", line 1008, in run_simple
    inner()
  File "/datasets/rzamora/miniconda3/envs/debug_8072/lib/python3.8/site-packages/werkzeug/serving.py", line 948, in inner
    srv = make_server(
  File "/datasets/rzamora/miniconda3/envs/debug_8072/lib/python3.8/site-packages/werkzeug/serving.py", line 780, in make_server
    return ThreadedWSGIServer(
  File "/datasets/rzamora/miniconda3/envs/debug_8072/lib/python3.8/site-packages/werkzeug/serving.py", line 686, in __init__
    super().__init__(server_address, handler)  # type: ignore
  File "/datasets/rzamora/miniconda3/envs/debug_8072/lib/python3.8/socketserver.py", line 452, in __init__
    self.server_bind()
  File "/datasets/rzamora/miniconda3/envs/debug_8072/lib/python3.8/http/server.py", line 138, in server_bind
    socketserver.TCPServer.server_bind(self)
  File "/datasets/rzamora/miniconda3/envs/debug_8072/lib/python3.8/socketserver.py", line 466, in server_bind
    self.socket.bind(self.server_address)
OSError: [Errno 98] Address already in use
```

Reply:
Okay - I was able to confirm that `dd.utils.assert_eq` should work fine in the two parquet/dataframe tests here.